src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1287 @@
+                with vfs.open(f, "rb") as src:
+                    # only open header
+                    las = laspy.open(src)
+                    hdr = las.header
+                    if mins is None:
+                        mins = hdr.mins
+                    else:
+                        mins = [min(e) for e in zip(mins, hdr.mins)]
+
+                    if maxs is None:
+                        maxs = hdr.maxs
+                    else:
+                        maxs = [max(e) for e in zip(maxs, hdr.maxs)]
+
+                    if scales is None:
+                        scales = hdr.scales
+                    else:
+                        scales = [max(e) for e in zip(scales, hdr.scales)]
+
+                    if offsets is None:
+                        offsets = hdr.offsets
+                    else:
+                        offsets = [min(e) for e in zip(offsets, hdr.offsets)]
+
+            return GeoMetadata(
+                paths=sources,
+                extents=BoundingBox(
+                    minx=mins[0],
+                    miny=mins[1],
+                    minz=mins[2],
+                    maxx=maxs[0],
+                    maxy=maxs[1],
+                    maxz=maxs[2],
+                ),
+                scales=tuple(scales),
+                offsets=tuple(offsets),
+            )
+
+
+def get_geometry_metadata(
+    sources: Iterable[os.PathLike],
+    config: Mapping[str, Any] = None,
+    verbose: bool = False,
+    id: str = "pointcloud_metadata",
+    trace: bool = False,
+    log_uri: Optional[str] = None,
+) -> GeoMetadata:
+    """Return geospatial metadata for a sequence of input geometry data files
+
+    :param sources: A sequence of paths or path to input
+    :param config: dict configuration to pass on tiledb.VFS
+    :param verbose: bool, enable verbose logging, default is False
+    :param trace: bool, enable trace logging, default is False
+    :param log_uri: Optional[str] = None,
+    :Return: GeoMetadata, a populated GeoMetadata object
+    """
+    # note this will be refactored to use /vsipyopener in fiona
+    if len(sources) == 0:
+        raise ValueError("Input point cloud datasets required")
+
+    with tiledb.scope_ctx(config):
+        tiledb.VFS()
+        logger = get_logger_wrapper(verbose)
+        with Profiler(array_uri=log_uri, id=id, trace=trace):
+            # for geometries we need the maximum extents
+            # and to check geometry types / crs are the same
+            schema = None
+            full_extents = None
+            crs = None
+
+            for pth in sources:
+                logger.debug("finding metadata for %r", pth)
+                with fiona.open(pth) as src:
+                    if crs is None:
+                        crs = src.crs
+                    else:
+                        if src.crs != crs:
+                            raise ValueError("All datasets need to be in the same CRS")
```

Comment:
This can be shortened a bit:

```python
crs = crs or src.crs
# Alternately, if `""` or some other falsey value is a valid crs or schema:
crs = src.crs if crs is None else crs
if src.crs != crs:  # this can now run unconditionally
  raise ...
```